displacy/displacy_service/server.py

```diff
@@ -108,8 +108,20 @@ class DepResource(object):
 
     test with: curl -s localhost:8000/dep -d '{"text":"Pastafarians are smarter than people with Coca Cola bottles."}'
     """
+
+    @staticmethod
+    def pred(key, valid_keys):
+        return valid_keys is None or key in valid_keys
+
+    def dep_text(self, text, model, collapse_punctuation, collapse_phrases, **params):
+        parse = Parse(model, text, collapse_punctuation, collapse_phrases).to_json()
+        arcs = parse['arcs']
+        words = [{k:v for k,v in word.items() if self.pred(k, params.get('features', None))} for word in parse['words']]
```

Comment:
You'll have to do something fancy with vector (for an example of "something", see #1 .) It needs to be serialised to be passed over the wire, and it's best to use the internal numpy serialisation to avoid loss of precision.

Reply:
(this is following a discussion we had in realspace)

we _do_ utilize the parse logic from #1 on L117 where we instantiate a `Parse` obj and call `to_json` on it in a one-liner

Reply:
(this is following a discussion we had in realspace)

we _do_ utilize the parse logic from #1 on L117 where we instantiate a `Parse` obj and call `to_json` on it in a one-liner